deepchecks/base/check.py

```diff
@@ -165,6 +189,72 @@ def display_check(self, unique_id: str = None, as_widget: bool = False,
             return box
         display_html(check_html, raw=True)
 
+    def _display_to_json(self):
+        displays = []
+        old_backend = matplotlib.get_backend()
+        displays.append(('header', f'<h4>{self.get_header()}</h4>'))
+        displays.append(('docs_header', f'<p>{get_docs_summary(self.check)}</p>'))
+        if self.conditions_results:
+            displays.append(('conditions_header', _CONDITIONS_HEADER))
+            cond_df = get_conditions_table(self)
+            if isinstance(cond_df, Styler):
+                cond_df = cond_df.data
+            displays.append(('conditions', cond_df.to_json(orient='records')))
+        displays.append(('outputs_header', _ADDITIONAL_OUTPUTS_HEADER))
+        for item in self.display:   
+            if isinstance(item, Styler):
+                displays.append(('dataframe', item.data.to_json(orient='records')))
+            elif isinstance(item, pd.DataFrame):
+                displays.append(('dataframe', item.to_json(orient='records')))
+            elif isinstance(item, str):
+                displays.append(('html', item))
+            elif isinstance(item, BaseFigure):
+                displays.append(('plotly', item.to_json()))
+            elif callable(item):
+                try:   
+                    matplotlib.use('Agg')
+                    item()
+                    displays.append(('plt', _save_all_open_figures()))
+                except Exception:
+                    displays.append(('plt', ''))
+            else:
+                matplotlib.use(old_backend)
+                raise Exception(f'Unable to create json for item of type: {type(item)}')
+        matplotlib.use(old_backend)
+        return displays
+
+    def to_json(self, with_display: bool = True):
+        check_name = self.check.name()
+        parameters = self.check.params()
+        result_json = {'name': check_name, 'parameters': parameters}
+        
+        if isinstance(self.value, pd.DataFrame):
+            result_json['value'] = self.value.to_json()
+        elif isinstance(self.value, np.ndarray):
+            result_json['value'] = self.value.tolist()
+        else:
+            result_json['value'] = self.value
```

Comment:
what will happen with dict that contains dataframe/ndarray?

Reply:
jsonpickle will handle it, though now it isn't happening. I can't go into dicts recursivly